### PR TITLE
Tests for extended cpaper example and improved docs

### PR DESCRIPTION
**Pull request by TopHatCroat**
Hi, this seems like a very useful repo so I wanted to contribute a bit.

I think, since you already have 2 commercial paper examples, that one should be a close copy to the original example, so that people can see how easy it can be done using this lib, and the other should have a bit more features and showcase the goodies this lib provides.

I hope you agree and please do nitpick :smile:

I would also like to update the main README in here with a bit more user friendly examples as currently it's a bit hard to follow.

### CODE REVIEW

**Comment by vitiko** [comment]
Great idea to separate base commercial paper example with struct based schema and extented example with protobuf state etc.
But at the moment mapping tests use schema from `cpaper/schema` package https://github.com/s7techlab/cckit/blob/master/state/mapping/mapping_test.go, thus tests are broken. Can you modify this tests to use schema only from `cpaper_extended` example?

**Comment by TopHatCroat** [comment]
Yep, my bad, I forgot to run the full test suite before submitting.

Though I think it would be much better to create new proto files (probably just c/p them from cpaper_extended example) in the `cckit/state/mapping/testdata/schema/
` directory and use those for mapping tests because then it will be easier to create new tests for any further mapping features without worrying about breaking example tests and vice versa.

**Comment by TopHatCroat** [comment]
So I've added new tests which are separated from the example chaincode as I suggested. I think this way it will be better to work with in the future.
But if you think this adds too much code than it's worth I'll revert it back and make the tests work with the examples.
